devicedefender/include/aws/iotdevicedefender/DeviceDefender.h

```diff
@@ -98,6 +203,10 @@ namespace Aws
                 void *cancellationUserdata = nullptr) noexcept;
 
             static void s_onDefenderV1TaskCancelled(void *userData);
+
+            // Holds all of the custom metrics created for this task. These are pointers that will be
+            // automatically created and cleaned by ReportTask when it is destroyed.
+            Crt::Vector<std::shared_ptr<CustomMetricBase>> storedCustomMetrics;
```

Comment:
You could better insulate the type definitions by forward declaring CustomMetricBase and then moving its definition plus the definitions of all subclasses to the .cpp file.  The function typedefs would of course stay here.